src/scripts/browserDeviceProfile.js

```diff
@@ -942,8 +951,14 @@ export default function (options) {
         av1VideoRangeTypes += '|HLG';
     }
 
-    if (supportsDolbyVision(options) && canPlayDolbyVisionHevc(videoTestElement)) {
-        hevcVideoRangeTypes += '|DOVI';
+    if (supportsDolbyVision(options)) {
+        const profiles = supportedDolbyVisionProfilesHevc(videoTestElement);
+        if (profiles.includes(5) || profiles.includes(7)) {
```

Comment:
Profile 7 is backward compatible with HDR10.

Reply:
> Maybe, but i don't think we are tagging it correctly anyways. Server-side we are only handling cross-compatibility IDs 1 and 4 separately and the rest are being classified as "DOVI". As far as I can tell from dolby's pdf, profile 7 usually has a cross-compatibility ID of 6, for which we do nothing special.

Indeed, I seem to have missed this. We should label this case as `DOVIWithHDR10`. Moreover, we only add `DOVI*` for profiles that support HDR. Profiles that only support SDR are ignored. More accurately we should annotate `VideoRangeType.DOVI` in the server as a type that does not contain any fallback layer.

> 1.9. * Profile and Level for Dolby Vision MUST be Profile 5 (single layer 10-bit HEVC) and less than or equal to Level 7.

> 1.9b. + For maximum compatibility, some Dolby Vision variants SHOULD be less than or equal to Level 7.

We use the HLS documentation [provided by Apple](https://developer.apple.com/documentation/http-live-streaming/hls-authoring-specification-for-apple-devices#Video) as a reference. Profile 7 over fMP4-HLS is supported.

Reply:
I did more research and found that usually DV Profile 7 videos are not tagged as `dvh1` but as `hvc1`. When Profile 7 needs to be streamed via HLS, it needs to be converted to Profile 8 to ensure compatibility. So instead of adding a new enum, we force Profile 7 to be marked as hvc1 because currently FFmpeg does not have the ability to convert DV Profiles.